tests/e2e/asserter.ml

```diff
@@ -0,0 +1,44 @@
+open Cmdliner
+open Bin_common
+open Crypto
+
+let assert_state_correct data_folder expected_state_root_hash
+    expected_block_height =
+  print_endline @@ "Checking state at path: " ^ data_folder;
+  let file = data_folder ^ "/state.bin" in
+  let protocol_state = Lwt_main.run @@ Files.State_bin.read ~file in
+  Format.printf "Minimum expected block height: %Ld. Actual height: %Ld\n%!"
+    expected_block_height protocol_state.block_height;
+  assert (protocol_state.block_height >= expected_block_height);
+  let actual_state_root_hash =
+    protocol_state.state_root_hash |> BLAKE2B.to_string in
+  Format.printf "Expected state root hash: %s. Actual state root hash: %s\n%!"
+    expected_state_root_hash actual_state_root_hash;
+  assert (actual_state_root_hash = expected_state_root_hash);
+  print_endline "State looks good 👍"
+(* TODO: add more assertions *)
+
+let args =
```

Comment:
Can we move the 3 functions `folder_node, expected_state_root_hash, expected_block_height` outside, and before that open `Arg`, like this:
```
open Arg

let folder_node = ...
let expected_state_root_hash = ...
let expected_block_height = ...

let args =
let open Term in
const assert_state_correct ..
```